pkg/acpi/rsdp.go

```diff
@@ -18,7 +18,7 @@ var (
 // RSDP is the v2 version of the ACPI RSDP struct.
 type RSDP struct {
 	// base is the base address of the RSDP struct in physical memory.
-	base uint64
+	base int64
```

Comment:
why the change to int64? it's a pointer, right?

Reply:
it's a physical address, and since things like PreadAt and Seek want int64 I find the consistency preferable. It's not a pointer in a reasonable sense of the word.

Reply:
again on this, I'm not that tied to it, but a uint64 with the high order bit set will flat out not work with again kernel device, esp. /dev/mem, so I started thinking I made I mistake first time I wrote this code for uint64.

Reply:
int64 sounds fair in that context

Reply:
I thought some more. Look at it this way. It's not a pointer, it's a physical address. The difference? A pointer is a virtual concept, tied to address mapping hardware. It gives you pointers that can't exist in the real world. A physical address is what the hardware can do. amd64 has had over the years 43 bit, 48 bit, and maybe, someday, 53 bit PAs. I don't think we'll see 64-bit PAs until we get to 128-bit systems, at which point most all of this will be changing anyway. 

barret had better code: https://play.golang.org/p/4ZBZwFLzh_d 

all that said, fun discussion, if you feel uint64 is best I'm fine with it.

or put another way, the 64-bit pointers in acpi are populated on 32-bit kernels, so the machine with 43-bit PAs is running with 32-bit pointers, and has mappings from one to the other.

Reply:
You've already got me, no need to debate it more :)